README.md

```diff
@@ -251,6 +251,16 @@ export TRITON_OVERRIDE_DIR=<override_dir>
 # Step 4: Run the kernel again to see the overridden result
 ```
 
+**Compiler Pipeline Override Steps**
+```bash
+export TRITON_ALWAYS_COMPILE=1
+export TRITON_DUMP_PASS_STAGES=1
+export TRITON_OVERRIDE_PASS_STAGES=1
+# Step 1: Run the kernel once to dump compiler pipe's compiler_override.py file
+# Step 2: Modify the pass pipeline you do want to override (delete, add, reorder)
+# Step 3: Run the kernel again to see the overridden result
+# Currently only over ridding the TTIR and TTGIR pipeline is supported
```

Comment:
I'm a bit confused on the use case, is it meant to be for debugging or for downstream plugins? 
I thought the added was going to be a downstream plugin which could just be a hook but it looks like this is setup for debugging

Reply:
Yeah, that is the use case we have in mind. @Jokeren has a few other use cases for custom instrumentation that would insert some analysis passes (add, disable, rearrange existing passes)